SF iOS/SF iOS/Helpers/UserNotification/BackgroundFetchScheduler.m

```diff
@@ -0,0 +1,88 @@
+//
+//  BackgroundFetchScheduler.m
+//  Coffup
+//
+//  Created by Zachary Drayer on 6/3/19.
+//
+
+#import "BackgroundFetchScheduler.h"
+#import "GroupBackgroundFetcher.h"
+
+#if defined(__IPHONE_13_0)
+@import BackgroundTasks;
+#endif
+
+@import UIKit;
+
+static NSString *const BackgroundTaskGroupUpdateIdentifier = @"BackgroundTaskGroupUpdateIdentifier";
+
+@interface BackgroundFetchScheduler ()
+
+@property (nonatomic) GroupBackgroundFetcher *bgFetcher;
+
+@end
+
+@implementation BackgroundFetchScheduler
+
+- (void)schedule
+{
+#if defined(__IPHONE_13_0)
+	if (@available(iOS 13.0, *)) {
+		// cancel existing task
+		[BGTaskScheduler.sharedScheduler cancelTaskRequestWithIdentifier:BackgroundTaskGroupUpdateIdentifier];
+
+		// re-schedule it
+		BGAppRefreshTaskRequest *taskRequest = [[BGAppRefreshTaskRequest alloc] initWithIdentifier:BackgroundTaskGroupUpdateIdentifier];
+		taskRequest.earliestBeginDate = [NSDate dateWithTimeIntervalSinceNow:60 * 60]; // wait at least an hour
+
+		NSError *taskSubmissionError = nil;
+		BOOL taskSubmitted = [BGTaskScheduler.sharedScheduler submitTaskRequest:taskRequest error:&taskSubmissionError];
+		if (!taskSubmitted) {
+			// oops
+		}
+	} else {
+#endif
+		[UIApplication.sharedApplication setMinimumBackgroundFetchInterval:UIApplicationBackgroundFetchIntervalMinimum];
+#if defined(__IPHONE_13_0)
+	}
+#endif
+}
+
+- (void)registerLaunchHandlers
+{
+#if defined(__IPHONE_13_0)
+	if (@available(iOS 13.0, *)) {
+		[BGTaskScheduler.sharedScheduler registerForTaskWithIdentifier:BackgroundTaskGroupUpdateIdentifier
+															usingQueue:dispatch_get_main_queue()
```

Comment:
Do we really need the main queue here?

Reply:
we can pass in `nil` and it will pick an arbitrary background q, but, since this is running in the background, main q seems fine?